solr/solrj/src/java/org/apache/solr/common/cloud/CollectionSetWatcher.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.common.cloud;
+
+import java.util.Set;
+
+/**
+ * Callback registered with {@link ZkStateReader#registerCollectionSetWatcher(CollectionSetWatcher)}
+ * and called whenever the cluster's collection set changes.
+ */
+public interface CollectionSetWatcher {
+
+  /**
+   * Called when a collection is created, a collection is deleted or a watched collection's state changes.
+   *
+   * Note that, due to the way Zookeeper watchers are implemented, a single call may be
+   * the result of several collection set changes. Also, multiple calls to this method can be made
+   * with the same colllection set, ie. without any new updates.
+   *
+   * @param collections       the set of collections
+   *
+   * @return true if the watcher should be removed
```

Comment:
What's the case here for removing a watcher, and why is `true` the best option? I'm not totally against it, but a boolean seems an odd choice here. If boolean is used would returning `true` to continue watching be a better choice?

Reply:
This follows the convention used in `CollectionStateWatcher`.